shell/platform/darwin/macos/framework/Source/FLEViewController.mm

```diff
@@ -472,6 +492,26 @@ - (void)dispatchKeyEvent:(NSEvent*)event ofType:(NSString*)type {
   }];
 }
 
+- (void)onSettingsChanged:(NSNotification*)notification {
+  NSString* brightness =
+      [[NSUserDefaults standardUserDefaults] stringForKey:@"AppleInterfaceStyle"];
+  [_settingsChannel sendMessage:@{
+    @"platformBrightness" : [brightness isEqualToString:@"Dark"] ? @"dark" : @"light",
+    // TODO(jonahwilliams): https://github.com/flutter/flutter/issues/32006.
+    @"textScaleFactor" : @1.0,
+    @"alwaysUse24HourFormat" : @false
+  }];
+}
+
+- (void)sendInitialSettings {
+  [[NSDistributedNotificationCenter defaultCenter]
```

Comment:
And based on https://developer.apple.com/documentation/appkit/supporting_dark_mode_in_your_interface?language=objc a TODO to switch this to KVO on `effectiveAppearance`